pkg/terraform/parser/parser.go

```diff
@@ -159,31 +160,59 @@ func (p *TerraformParser) ProcessStates(ctx context.Context, gcsUris []string) (
 		if err := json.NewDecoder(lr).Decode(&state); err != nil {
 			return nil, fmt.Errorf("failed to decode terraform state: %w", err)
 		}
-		iams = append(iams, p.parseTerraformStateIAM(ctx, state)...)
+		parsedIAM, err := p.parseTerraformStateIAM(ctx, state)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode terraform state: %w", err)
+		}
+		iams = append(iams, parsedIAM...)
 	}
 	return iams, nil
 }
 
-func (p *TerraformParser) parseTerraformStateIAM(ctx context.Context, state TerraformState) []*assetinventory.AssetIAM {
+func (p *TerraformParser) parseTerraformStateIAM(ctx context.Context, state TerraformState) ([]*assetinventory.AssetIAM, error) {
 	var iams []*assetinventory.AssetIAM
 	for _, r := range state.Resources {
+		targetResources := []string{
+			"google_organization_iam_binding",
+			"google_folder_iam_binding",
+			"google_project_iam_binding",
+			"google_organization_iam_member",
+			"google_folder_iam_member",
+			"google_project_iam_member",
+		}
+
+		// short circuit if we dont find a type we want
+		if !slices.Contains(targetResources, r.Type) {
+			continue
+		}
+
+		raw, err := json.Marshal(r.Instances)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode terraform state: %w", err)
+		}
+
+		instances := make([]ResourceInstance, len(r.Instances))
+		if err := json.Unmarshal(raw, &instances); err != nil {
+			return nil, fmt.Errorf("failed to decode terraform state: %w", err)
+		}
```

Comment:
What are you trying to do?

Reply:
Were parsing into a slice of any (https://github.com/abcxyz/guardian/pull/318/files#diff-06c48a637212ad3944309b75d80941c60f39c29fc99da7d27332e59aa0e580faR59).

I want to now type cast that slice into []ResourcesInstances (because I have determine these are the resources I care out and have the expected attributes). I dont think you can cast, so I am marshalling and unmarshalling to parse into the expected struct type.

Reply:
@sethvargo figured it out, if we use json.RawMessage (equivalent to []byte) we can just unmarshall into the expected struct.